Sources/NFCPassportReader/PassportReader.swift

```diff
@@ -197,10 +200,16 @@ extension PassportReader : NFCTagReaderSessionDelegate {
             } catch let error as NFCPassportReaderError {
                 let errorMessage = NFCViewDisplayMessage.error(error)
                 self.invalidateSession(errorMessage: errorMessage, error: error)
-            } catch let error {
-                Log.debug( "tagReaderSession:failed to connect to tag - \(error.localizedDescription)" )
-                let errorMessage = NFCViewDisplayMessage.error(NFCPassportReaderError.ConnectionError)
-                self.invalidateSession(errorMessage: errorMessage, error: NFCPassportReaderError.Unknown(error))
+            } catch {
+                if let nfcError = error as? NFCReaderError {
+                    if nfcError.errorCode == NFCReaderError.readerTransceiveErrorTagResponseError.rawValue {
+                        let errorMessage = NFCViewDisplayMessage.error(NFCPassportReaderError.ConnectionError)
+                        self.invalidateSession(errorMessage: errorMessage, error: NFCPassportReaderError.ConnectionError)
+                    }
+                } else {
```

Comment:
What error case is this exactly? 🤔

Reply:
Good question. The explanation of this error is:

```
// The tag has responded with an error.
static var readerTransceiveErrorTagResponseError: NFCReaderError.Code 
```

However, the following error would **suit better** according to its description:

```
// The reader lost the connection to the tag.
static var readerTransceiveErrorTagConnectionLost: NFCReaderError.Code

```

The NFCReaderError from the first code block was thrown by the CoreNFC lib when the connection was lost to the chip. This is why I chose this error.

Reply:
_We discussed offline:_

- `readerTransceiveErrorTagResponseError` is thrown when we force a "connection lost" scenario: moving the phone away from the passport. Let's document this inline though since it is not clear just from reading the code.
- Also include `NFCReaderTransceiveErrorTagConnectionLost` in the mapping (even though it might not be reproducible for now but from the naming it is quite clear that it also makes sense to map to a `NFCPassportReaderError.ConnectionError`).